src/services/recim/match.ts

```diff
@@ -123,6 +123,13 @@ const updateMatch = (ID: number, updatedMatch: PatchMatch): Promise<CreatedMatch
 const deleteMatchCascade = async (matchID: number): Promise<CreatedMatch> =>
   http.del(`recim/matches/${matchID}`);
 
+// shortcut service to delete a list of matches
+const deleteMatchList = async (matches: Match[]) => {
+  for (let i = 0; i < matches.length; i++) {
+    await http.del(`recim/matches/${matches[i].ID}`);
+  }
+};
```

Comment:
This will be MUCH more performant without the inner `await`, since you can send all the requests in parallel instead of awaiting each request before sending the next one.

Also, generally better to use a `for in` loop to iterate arrays or else `Array.map`, since plain for loops clutter the important logic and introduce extra room for errors:
```suggestion
const deleteMatchList = (matches: Match[]) => 
  Prmoise.all(matches.map(match => deleteMatchCascade(match.ID)));
```

Reply:
Unfortunately we can't actually do this, I wanted to mess with cleaner implementations of javascript (I don't think I have those commits) but the main issue is that our actual `delete` route, since it also patches all associated sub resources i.e Activity -> series -> team/match, we HAVE to await each call or else under load (even just 15-20 matches) there are 500 errors thrown all over the place.

Reply:
Hmm, I'm looking at the code for `MatchService.DeleteMatchCascadeAsync` that is called by this route in the backend, and all that you're doing there is setting the status of this match and all of it's Match Team records to Deleted. That shouldn't cause any Database conflicts.

Reply:
This is not a blocking issue, so it doesn't need to be something you investigate deeply right now, I just am not sure how it could be causing the problems you described.